### PR TITLE
refactor: remove delete predicates from read filter interface

### PR DESCRIPTION
**Pull request by tustvold**
There is more cleanup that can be done here, but wanted to get sign off on the change to the public API.

QueryChunk is the interface for the query engine, it is implemented by DbChunk which is a snapshot of a catalog chunk, along with a test implementation. Given the catalog chunk is the owner of the delete predicates, it therefore seems strange that you would need to provide them to the QueryChunk? In my opinion they are an implementation detail of how the specific implementation of QueryChunk decides to handle deletes.

This PR therefore removes this from the interface.

### CODE REVIEW

File: query/src/provider/physical.rs [hunk 1]
```diff
@@ -118,22 +118,14 @@ impl<C: QueryChunk + 'static> ExecutionPlan for IOxReadFilterNode<C> {
...
-            .read_filter(&self.predicate, selection, &del_preds)
-            .map_err(|e| {
-                DataFusionError::Execution(format!(
-                    "Error creating scan for table {} chunk {}: {}",
-                    self.table_name,
-                    chunk.id(),
-                    e
-                ))
-            })?;
+        let stream = chunk.read_filter(&self.predicate, selection).map_err(|e| {
```
Comment (e-dard) [inline query/src/provider/physical.rs:121]: I think this code highlights the deletes leaking unnecessarily out of the chunk abstraction

**Review by e-dard — APPROVED** [review]
This change seems pretty sane to me. Would be good to get @NGA-TRAN's sign off because she has been working on this front and may have reasons for the way the code is organised.

File: query/src/provider/physical.rs (hunk 1 above)
Comment (alamb) [reply query/src/provider/physical.rs:121]: As in this PR reduces the leakage or that the PR increases the leakage?

**Review by alamb — APPROVED** [review]

File: query/src/provider/physical.rs (hunk 1 above)
Comment (e-dard) [reply query/src/provider/physical.rs:121]: Reduces. Before, the delete predicates were being fetched out of the chunk only to be put back into the chunk's `read_filter` implementation.

Comment (alamb) [reply query/src/provider/physical.rs:121]: Yes I agree this PR encapsulates the deletes more

File: server/src/db/chunk.rs [hunk 2]
```diff
@@ -346,19 +346,23 @@ impl QueryChunk for DbChunk {
...
         debug!(?predicate, "Input Predicate to read_filter");
         self.access_recorder.record_access();
 
-        debug!(?delete_predicates, "Input Delete Predicates to read_filter");
+        let delete_predicates: Vec<_> = self
+            .meta
+            .delete_predicates
+            .iter()
+            .map(|pred| Arc::new(pred.as_ref().clone().into()))
+            .collect();
```
Comment (NGA-TRAN) [inline server/src/db/chunk.rs:361]: 👍

**Review by NGA-TRAN — COMMENTED** [review]
Great catch and thanks for he cleanup, @tustvold

**Pull request merged**